profile_page/database/profile_page_db.py

```diff
@@ -169,108 +158,162 @@ def find_user_game(user_id : str, user_game_id: str):
+        "badges":[],
+        "current_story_level":1,
+        "friends":[],
+        "points":0,
+        "rank":"Baby Panda",
+        "save_games":[],
+        "sharing": True
+        }
+
+        return save_user(user)
 
-def update_user_game_graph(user_id : str, user_game_id: str, graph_data: dict):
+def login ( user_id : str, passhash : str ):
     """
-    Finds an active user in the Database and updates a specific game's graph
+    Validates a user's password in the database
 
     Parameters:
     user_id (str): unique identification for the user
-    game_index (int): locates which save game to overwrite
-    graph_data (dictionary): game board graph dictionary
+    passhash (str): hashed string representing the user's password
 
     Returns:
     On Success:
-        dictionary: original user game
+        Boolean: True
     On Fail:
-        str: friendly response to infrom of an error
+        Boolean: False
     """
-    value_returned = client.InitialDB.User_Profile.find_one_and_update(
+
+    value_returned = client.InitialDB.User_Profile.find({"$and": [{ "user_id": user_id}, {"password_hash": passhash}]}).count()
+    return value_returned > 0
+
+def update_token( user_id : str, token : str ):
+    """
+    Updates a user's API token in the database
+
+    Parameters:
+    user_id (str): unique identification for the user
+    token (str): authentication for API calls
+
+    Returns:
+    On Success:
+        Boolean: True
+    On Fail:
+        Boolean: False
+    """
+    value_returned = client.InitialDB.User_Profile.update_one(
         {"user_id": user_id},
-        {"$set":{"save_games.$[index].graph":graph_data}},
-        upsert=False, return_document = ReturnDocument.AFTER,
-        array_filters=[{"index.game_id":{"$eq": user_game_id}}]
-    )
+        {"$set":{"auth_token":token}},
+        upsert=False
+    ).modified_count
 
-    if value_returned == None:
-        return 'nah bro idk about it'
-    return value_returned
+    return value_returned > 0
 
-def find_user_game_graph(user_id : str, user_game_id: str):
+def check_user( user_id : str, token : str ):
     """
-    Finds an active user in the Database and returns a specific game to the API
+    Validates a user's API token in the database
 
     Parameters:
     user_id (str): unique identification for the user
-    game_index (int): locates which save game to overwrite
+    token (str): authentication for API calls
 
     Returns:
     On Success:
-        dictionary: original user game
+        Boolean: True
     On Fail:
-        str: friendly response to infrom of an error
+        Boolean: False
     """
+    value_returned = client.InitialDB.User_Profile.find({"$and": [{ "user_id": user_id}, {"auth_token": token}]}).count()
+    return value_returned > 0
 
-    value_returned = client.InitialDB.User_Profile.find_one(
-        {"user_id": user_id},
-        {'_id':0,"save_games":1}
-    )
+def remove_token ( user_id : str ):
```

Comment:
Let's add a boolean flag (and unit test) so that null (None) user token can't authenticate.